### ISSUE TITLE
Customizing the Neural Network

### ISSUE THREAD

**Issue by shreyk25** [NVlabs/tiny-cuda-nn#106]
Hi, would it be possible to have different number of neurons in each hidden layer?
Also, it would be great if you could share a replacement to nn.Linear function in tiny-cuda-nn (if it is at all possible). I have a very custom neural network and the simple parameters available for the network might not work for me.
Thanks!

**Comment by Tom94** [comment]
Hi there. No, customizing the neural network architecture beyond what's offered in the json docs is not possible with __tiny-cuda-nn__ and currently not on the roadmap.

PyTorch's `nn.Linear` layer already has pretty much as good a performance as you can get if you don't fuse, so offering a replacement at that granularity wouldn't make much sense either. For your use-case, you're probably best off by using __tiny-cuda-nn__ only for its input encoding. Cheers!

**Comment by shreyk25** [comment]
To give you more details, i would like to make the model faster using tiny-cuda-nn.

`class NeRF(nn.Module):
    def __init__(self, D=8, W=256, input_ch=3, input_ch_views=3, output_ch=4, skips=[4], use_viewdirs=False):
        """ 
        """
        super(NeRF, self).__init__()
        self.D = D
        self.W = W
        self.input_ch = input_ch
        self.input_ch_views = input_ch_views
        self.skips = skips
        self.use_viewdirs = use_viewdirs
        
        self.pts_linears = nn.ModuleList(
            [nn.Linear(input_ch, W)] + [nn.Linear(W, W) if i not in self.skips else nn.Linear(W + input_ch, W) for i in range(D-1)])
        
        ### Implementation according to the official code release (https://github.com/bmild/nerf/blob/master/run_nerf_helpers.py#L104-L105)
        self.views_linears = nn.ModuleList([nn.Linear(input_ch_views + W, W//2)])

        ### Implementation according to the paper
        # self.views_linears = nn.ModuleList(
        #     [nn.Linear(input_ch_views + W, W//2)] + [nn.Linear(W//2, W//2) for i in range(D//2)])
        
        if use_viewdirs:
            self.feature_linear = nn.Linear(W, W)
            self.alpha_linear = nn.Linear(W, 1)
            self.rgb_linear = nn.Linear(W//2, 3)
        else:
            self.output_linear = nn.Linear(W, output_ch)

    def forward(self, x):
        if x.shape[-1] > self.input_ch + self.input_ch_views:
            input_pts, input_views, input_depth = torch.split(x, [self.input_ch, self.input_ch_views, 1], dim=-1)
        else:
            input_pts, input_views = torch.split(x, [self.input_ch, self.input_ch_views], dim=-1)

        h = input_pts
        for i, l in enumerate(self.pts_linears):
            h = self.pts_linears[i](h)
            h = F.relu(h)
            if i in self.skips:
                h = torch.cat([input_pts, h], -1)

        if self.use_viewdirs:
            alpha = self.alpha_linear(h)
            feature = self.feature_linear(h)
            h = torch.cat([feature, input_views], -1)
        
            for i, l in enumerate(self.views_linears):
                h = self.views_linears[i](h)
                h = F.relu(h)

            rgb = self.rgb_linear(h)
            outputs = torch.cat([rgb, alpha], -1)
        else:
            outputs = self.output_linear(h)

        return outputs `

Could you please let me know what my approach should be in this case (if at all it is possible)?
